maintenance/wikia/WikiFactoryVariables/migrateWikiWordmarks.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+/**
+* Maintenance script to manage WikiFactory variable for one wiki or multiple wikis (from file)
+* - enable/disable the extension (set to true/false)
+* - set value to the variable (only string, integer, boolean types)
+* - remove variable from the wiki
+* - get the variable value
+* This is one time use script
+* @usage
+* 	# this will migrate wordmark-image-url for wiki with ID 119:
+* 	migrateWikiWordmarks -d --wikiId 119 --verbose --keyName wordmark-image-url
+*/
+
+ini_set( 'display_errors', 'stderr' );
+ini_set( 'error_reporting', E_NOTICE );
+
+require_once( dirname( __FILE__ ) . '/../../Maintenance.php' );
+
+use \Wikia\Logger\WikiaLogger;
+
+/**
+ * Class MigrateWikiWordmarks
+ */
+class MigrateWikiWordmarks extends Maintenance {
+
+	protected $dryRun  = false;
+	protected $verbose = false;
+	protected $keyName = '';
+	const WIKI_FACTORY_VARIABLE = "wgOasisThemeSettings";
+
+	public function __construct() {
+		parent::__construct();
+		$this->mDescription = "Migrates variables in wgOasisThemeSettings to HTTPS";
+		$this->addOption( 'dry-run', 'Dry run mode', false, false, 'd' );
+		$this->addOption( 'verbose', 'Show extra debugging output', false, false, 'v' );
+		$this->addOption( 'keyName', 'Key in WikiFactory variable which should be migrated to https', true, true, 'k' );
+		$this->addOption( 'reason', 'Reason to provide when setting a variable', false, true );
+	}
+
+	public function execute() {
+		global $wgCityId;
+		$this->dryRun  = $this->hasOption( 'dry-run' );
+		$this->verbose = $this->hasOption( 'verbose' );
+		$this->keyName = $this->getOption( 'keyName', '' );
+		$reason        = $this->getOption( 'reason' );
+
+		if ( empty( $this->keyName ) ) {
+			$this->error( "Error: Empty key name." . PHP_EOL );
+			return false;
+		}
+
+		$varData = (array) WikiFactory::getVarByName( self::WIKI_FACTORY_VARIABLE, $wgCityId, true );
+		if ( empty( $varData['cv_id'] ) ) {
+			$this->error( "Error: " . self::WIKI_FACTORY_VARIABLE . " not found." . PHP_EOL );
+			return false;
+		}
+
+		echo "Variable: " . self::WIKI_FACTORY_VARIABLE . " (Id: $varData[cv_id])" . PHP_EOL;
+		$this->debug( "Variable data: " . json_encode( $varData ) . PHP_EOL );
+
+		$wg = F::app()->wg;
+		$wg->User = User::newFromName( Wikia::BOT_USER );
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)